owncloudApp/src/main/java/com/owncloud/android/MainApp.java

```diff
@@ -65,17 +65,25 @@
     private static final String POLICY_SINGLE_SESSION_PER_ACCOUNT = "single session per account";
     @SuppressWarnings("unused")
     private static final String POLICY_ALWAYS_NEW_CLIENT = "always new client";
-    private static final int DEFAULT_0 = 0;
+    private static final int CLICKS_DEFAULT = 0;
 
     private static Context mContext;
 
-    private static boolean mDeveloper;
-
     public void onCreate() {
         super.onCreate();
-        MainApp.mContext = getApplicationContext();
 
-        readIsDeveloper();
+        mContext = getApplicationContext();
+
+        if (isDeveloper()) {
+            String dataFolder = getDataFolder();
+
+            // Set folder for store logs
+            Log_OC.setLogDataFolder(dataFolder);
+
+            Log_OC.startLogging(Environment.getExternalStorageDirectory().getAbsolutePath());
+            Log_OC.d(BuildConfig.BUILD_TYPE, "start logging " + BuildConfig.VERSION_NAME + " " +
```

Comment:
With your change, you miss the start log entry, when you start to become a developer with `readIsDeveloper()` triggered by 
`((MainApp) getApplication()).readIsDeveloper(); // read value to global variable`
in _Preferences.java_

This means no
* start log entry 
* you have to kill/restart app to start log

I introduced `readIsDeveloper()` to start immediate becoming a developer

Reply:
I see your point and just updated the code but `readIsDeveloper` is not a good name for that method so I renamed to `startLogIfDeveloper`.

About `((MainApp) getApplication()).readIsDeveloper()` call in _Preferences.java_, if the app is installed as debug, why should we still need to click the version - at least once - to call `((MainApp) getApplication()).readIsDeveloper()` and start logging even when the Logs option already appears in settings view? I think logging might start automatically when the app is built as debug.

Reply:
It does log automatic in debug
![image](https://user-images.githubusercontent.com/3314607/55285431-636b1600-538c-11e9-9920-6e81333c6b34.png)

I introduce a `mDeveloper` to not access Sharedpreferes (a XML, which has to be parsed) every time. Yes, with your way you have no such variable, but you increase the chance to fire a [strict mode](https://developer.android.com/reference/android/os/StrictMode) violation, which Google collects and uses it to rank the app.

I guess when you enable strict mode in this app, it will be not a nice surprise

Reply:
This is the reason why finishing contributions from other people is not a good idea - especially when the contributor is still active in the repository - , you know better than me what is needed here because it was your contribution. If we had added all these comments, suggestions and explanations as answers to my code review of your contribution, we would have saved a lot of time and maybe no change to your code would have been necessary.

Have a look at the steps we've been following so far to get this contribution merged:
1. You opened https://github.com/owncloud/android/pull/2439/ with your contribution.
2. I reviewed your contribution and suggested some changes.
3. You answered there with things like below and say you're not in the mod to do all suggestions to get merged.
![Screenshot 2019-04-01 at 09 41 01](https://user-images.githubusercontent.com/15261363/55311057-5587c480-5462-11e9-8c23-d5fa26402228.png)
4. We decide to close the initial PR and open a new one to include your contribution with some of the changes requested in the first PR.
5. You start to review the new PR and open discussions that should have taken place in the old PR.
6. I apply the changes requested here and my solution is almost the same as the one included in the initial PR. 

It does not make any sense honestly and IMHO this kind of situations do make the team go slower. I always appreciate comments and suggestions from everyone, but this is a waste of time - we are going forward and backward - and I feel like we can do this in a better way by giving proper arguments to not apply changes in the initial code review, so please let's follow that approach from now on, thanks.

Regarding this PR, I will change what needed and move it forward.